include/flatbuffers/flatbuffers.h

```diff
@@ -658,7 +657,7 @@ class DetachedBuffer {
         cur_(cur),
         size_(sz) {}
 
-  // clang-format off
+// clang-format off
```

Comment:
these `// clang-format off` indentation changes are unfortunate. The whole reason we have them is because clang-format doesn't support indented #ifdefs, and indenting them makes code more "scannable" because they don't interrupt the indentation of the code. This once more interrupts the regular indentation of the code.

I am not sure why it is happening, surely it knows how to indent `//` correctly? Is this a special case for `// clang-format off` ??

Reply:
Or maybe it now thinks that since its followed by a line with `#`, it must be in the first column? Maybe we (gasp) need a blank line between them? :(

Note in https://clang.llvm.org/docs/ClangFormatStyleOptions.html IndentPPDirectives there is `BeforeHash` we want, but supposedly it doesn't work yet.. https://bugs.llvm.org/show_bug.cgi?id=36019 https://bugs.llvm.org/show_bug.cgi?id=36020

Reply:
Appears the issue is a [bug in clang-format](https://bugs.llvm.org/show_bug.cgi?id=39280) with improper formatting of the "// clang-format off" line when it precedes a #. And yes, adding a blank line afterwards avoids the issue. And I noticed there are a bunch of instances already in the code with a blank line in that position.

A simple fix is to run clang-format twice, as it fixes it self!